docs/ReleaseNotes.md

```diff
@@ -1,6 +1,24 @@
 # Acuminator Release Notes
 This document provides information about fixes, enhancements, and key features that are available in Acuminator.
 
+## Acuminator 3.1
+Acuminator 3.1 includes the bug fixes and enhancements described in this section, as well as the features that have been implemented in previous versions.
+
+### Enhancements
+In Acuminator 3.1, the following enhancements have been implemented:
+ - The [PX1056](diagnostics/PX1056.md) diagnostic has been added to check and warn if a graph is created in the `IsActive` method in graph and DAC extension and the `IsActiveForGraph` method.
+ - Code snippets are now provided for generic event handlers and DAC fields. 
+ 
+### Fixed Bugs
+In this version of Acuminator, the following bugs have been fixed:
+ - Diagnostic suppression performed using Acuminator code fix did not work in Visual Studio 2022 because it relied on internal API that was changed in the new version of Visual Studio
+ - The [PX1072](diagnostics/PX1072.md) diagnostic displayed a warning for static methods while it should report only graph instance methods.
+ - The [PX1086](diagnostics/PX1086.md) diagnostic failed on expression that create collections when the braces were omitted.
+ - Acuminator failed when the PXGraph.InstanceCreated.AddHandler  method had a type parameter.
+ - The [PX1008](diagnostics/PX1008.md) diagnostic displayed a warning when capturing a reference to `this` in a delegate even in a class that was not a graph or a graph extension.
```

Comment:
The related changes from ATR-452 are not mentioned in enhancements, See all changes in the "Changes Summary" field in ATR-670. The important things:
```
- Extended design of syntax walkers for inter-procedural analysis, added extension points and new API.
- Added check for captured adapters to PX1008 check and improved diagnostic message
- Added support for PX1008 check in custom views and attributes

- PX1008 check is completely reworked. Now it is a data flow diagnostic that goes recursively into methods and checks how the data is transferred to them. It is based on new nested walkers design.

Here are some principles of the data flow analysis used by the main PX1008 walker.
We maintain a stack of non-capturable elements. Before we visit a methods from a method call recursively we calculate what can capture reference to graph/adapter at the call site. The diagnostic supports four options:
- We are at the top of call stack and inside an action handler. In this case adapter parameter can be captured.
- We are inside graph/graph extension and `this` is passed among the arguments
- We are inside recursive call and already have some non-capturable parameters that are used in call arguments.
- We are inside a local method and parameters from containing methods have some non-capturable elements

 There is a dedicated walker that checks for parameters and `this` captured in arguments expression.
Additionally, there is a walker that check for reassignments of parameters before the call site.
```

Also this can be in `Other things`:
```
- Introduced nullable reference code analysis and nullable annotations to parts of Acuminator 
```